src/openrct2/world/Map.cpp

```diff
@@ -1832,44 +1834,82 @@ static uint8 map_get_highest_land_height(sint32 xMin, sint32 xMax, sint32 yMin,
     return max_height;
 }
 
-static money32 raise_land(sint32 flags, sint32 x, sint32 y, sint32 z, sint32 ax, sint32 ay, sint32 bx, sint32 by, sint32 selectionType)
+
+static sint32 map_correct_corners(sint32 corner)
+{
+    if (corner < 0)
+    {
+        corner = MAP_SELECT_TYPE_CORNER_3;
+    }
+    return corner;
+}
+
+static money32 raise_land(sint32 flags, sint32 x, sint32 y, sint32 z, sint32 point_a_x, sint32 point_a_y, sint32 point_b_x, sint32 point_b_y, sint32 selectionType)
 {
     money32 cost = 0;
+    uint32  corners[2] = { MAP_SELECT_TYPE_EDGE_0, MAP_SELECT_TYPE_EDGE_0 };
 
-    if (selectionType < 0 || selectionType >= (sint32)Util::CountOf(tile_element_raise_styles))
+    if (selectionType >= MAP_SELECT_TYPE_EDGE_0 && selectionType <= MAP_SELECT_TYPE_EDGE_3) 
+    {
+        sint32 diff = EDGE_TO_CORNERS_ROTATION_OFFSET;
+        // We need to transform edge - check double corner operations; diff cannot be explicit as define due to some optimisations (different result)
+        // extract corners from edge:
+        corners[0] = map_correct_corners(selectionType - diff);
+        corners[1] = map_correct_corners(corners[0] - 1);
+    }
+    else
+    {
+        corners[0] = selectionType;
+    }
+
+    if (corners[0] < MAP_SELECT_TYPE_CORNER_0 || corners[0] > MAP_SELECT_TYPE_FULL ||
+        corners[0] >= sizeof(tile_element_raise_styles))
     {
         log_warning("Invalid selection type %d for raising land", selectionType);
         return MONEY32_UNDEFINED;
     }
 
-    if ((flags & GAME_COMMAND_FLAG_APPLY) && gGameCommandNestLevel == 1) {
+    if ((flags & GAME_COMMAND_FLAG_APPLY) && gGameCommandNestLevel == 1) 
+    {
         audio_play_sound_at_location(SOUND_PLACE_ITEM, x, y, z);
     }
 
     // Keep big coordinates within map boundaries
-    ax = std::max<decltype(ax)>(32, ax);
-    bx = std::min<decltype(bx)>(gMapSizeMaxXY, bx);
-    ay = std::max<decltype(bx)>(32, ay);
-    by = std::min<decltype(by)>(gMapSizeMaxXY, by);
+    point_a_x = std::max<decltype(ax)>(32, point_a_x);
+    point_b_x = std::min<decltype(bx)>(gMapSizeMaxXY, point_b_x);
+    point_a_y = std::max<decltype(bx)>(32, point_a_y);
+    point_b_y = std::min<decltype(by)>(gMapSizeMaxXY, point_b_y);
```

Comment:
`ax`, `bx`, `ax` and `ay` inside `decltype` are not renamed, probably because of the rebase.